Warning in new closure type

```
In file included from ../../gccrs/gcc/rust/backend/rust-compile.h:24,
                 from ../../gccrs/gcc/rust/rust-session-manager.cc:56:
../../gccrs/gcc/rust/backend/rust-compile-context.h: In member function ‘virtual void Rust::Compile::TyTyResolveCompile::visit(Rust::TyTy::ClosureType&)’:
../../gccrs/gcc/rust/backend/rust-compile-context.h:664:34: warning: unused parameter ‘type’ [-Wunused-parameter]
  664 |   void visit (TyTy::ClosureType &type) override { gcc_unreachable (); }
      |               ~~~~~~~~~~~~~~~~~~~^~~~

```

since

b7bb6a9832958abb076dcd9207c7457a7bec6201

--- a/gcc/rust/backend/rust-compile-context.h
+++ b/gcc/rust/backend/rust-compile-context.h
@@ -660,7 +660,7 @@ public:
     ctx->insert_compiled_type (type.get_ty_ref (), named_struct, &type);
   }
 
-  void visit (const TyTy::ClosureType &type) override { gcc_unreachable (); }
+  void visit (const TyTy::ClosureType &) override { gcc_unreachable (); }
 
 private:
   TyTyResolveCompile (Context *ctx, bool trait_object_mode)

